rclcpp_examples/src/topics/allocator_example.cpp

```diff
@@ -138,16 +153,18 @@ int main(int argc, char ** argv)
   using rclcpp::memory_strategies::allocator_memory_strategy::AllocatorMemoryStrategy;
   rclcpp::init(argc, argv);
 
-  auto node = rclcpp::Node::make_shared("allocator_example", false);
-  // To use intra-process, comment out the above line and uncomment the following:
-  /*
-  auto context = rclcpp::contexts::default_context::get_global_default_context();
-  auto ipm_state =
-    std::make_shared<rclcpp::intra_process_manager::IntraProcessManagerState<MyAllocator<>>>();
-  // Constructs the intra-process manager with a custom allocator.
-  context->get_sub_context<rclcpp::intra_process_manager::IntraProcessManager>(ipm_state);
-  auto node = rclcpp::Node::make_shared("allocator_example", true);
-  */
+  rclcpp::Node::SharedPtr node;
+
+  if (argc > 1 && strstr(argv[1], "intra") != NULL) {
```

Comment:
What is `strstr(argv[1], "intra") != NULL` checking for? Can't this be more specific?